src/formatDate.js

```diff
@@ -8,7 +8,40 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateInOldFormat = Array.from(fromFormat);
+  const newFormat = Array.from(toFormat);
+  const dateOnly = date.split(fromFormat[3]);
```

Comment:
The line `const dateOnly = date.split(fromFormat[3]);` assumes that the delimiter is always at the fourth position in `fromFormat`. This may not be correct for all date formats. Consider dynamically determining the delimiter based on the actual format provided.